src/azure-cli/azure/cli/command_modules/rdbms/_params.py

```diff
@@ -214,166 +214,249 @@ def _complex_params(command_group):
+
+        administrator_login_password_arg_type = CLIArgumentType(
+            options_list=['--admin-password', '-p'],
+            help='The password of the administrator. Minimum 8 characters and maximum 128 characters. '
+                 'Password must contain characters from three of the following categories: '
+                 'English uppercase letters, English lowercase letters, numbers, and non-alphanumeric characters.',
+            arg_group='Authentication'
+        )
+
+        database_name_arg_type = CLIArgumentType(
+            metavar='NAME',
+            options_list=['--database-name', '-d'],
+            id_part='child_name_1',
+            help='The name of the database to be created when provisioning the database server',
+            local_context_attribute=LocalContextAttribute(
+                name='database_name',
+                actions=[LocalContextAction.GET, LocalContextAction.SET],
+                scopes=['{} flexible-server'.format(command_group)]))
+
+        tier_arg_type = CLIArgumentType(
+            options_list=['--tier'],
+            help='Compute tier of the server. Accepted values: Burstable, GeneralPurpose, Memory Optimized '
+        )
+
+        sku_name_arg_type = CLIArgumentType(
+            options_list=['--sku-name'],
+            help='The name of the compute SKU. Follows the convention Standard_{VM name}. Examples: Standard_D4s_v3'
+        )
+
+        storage_gb_arg_type = CLIArgumentType(
+            type=int,
+            options_list=['--storage-size'],
+            help='The storage capacity of the server. Minimum is 32 GiB and max is 16 TiB.'
+        )
+
+        backup_retention_arg_type = CLIArgumentType(
+            type=int,
+            options_list=['--backup-retention'],
+            help='The number of days a backup is retained. Range of 7 to 35 days. Default is 7 days.',
+            validator=retention_validator
+        )
+
+        version_arg_type = CLIArgumentType(
+            options_list=['--version'],
+            help='Server major version.'
+        )
+
+        iops_arg_type = CLIArgumentType(
+            type=int,
+            options_list=['--iops'],
+            help='Number of IOPS to be allocated for this server. You will get certain amount of free IOPS based '
+                 'on compute and storage provisioned. The default value for IOPS is free IOPS. '
+                 'To learn more about IOPS based on compute and storage, refer to IOPS in Azure Database for MySQL Flexible Server'
+        )
+
+        auto_grow_arg_type = CLIArgumentType(
+            arg_type=get_enum_type(['Enabled', 'Disabled']),
+            options_list=['--storage-auto-grow'],
+            help='Enable or disable autogrow of the storage. Default value is Enabled.'
+        )
+
+        yes_arg_type = CLIArgumentType(
+            options_list=['--yes', '-y'],
+            action='store_true',
+            help='Do not prompt for confirmation.'
+        )
```

Comment:
There's no need to implement double check arg by yourself.
You can use `confirmation` for command definition. See examples here:
https://github.com/Azure/azure-cli/blob/0850b5f713db57a921e5daf115fbfdffac5f7f31/src/azure-cli/azure/cli/command_modules/storage/commands.py#L108
https://github.com/Azure/azure-cli/blob/0850b5f713db57a921e5daf115fbfdffac5f7f31/src/azure-cli/azure/cli/command_modules/storage/commands.py#L119-L126